src/lib.rs

```diff
@@ -8,17 +8,17 @@ use std::thread;
 use std::sync::atomic::AtomicBool;
 use std::sync::atomic::Ordering as AtomicOrdering;
 use std::sync::{Arc, Mutex, Condvar};
-use std::sync::mpsc::{channel, Sender};
+use std::sync::mpsc::{channel, Receiver, Sender};
 use std::collections::BinaryHeap;
 use chrono::{Duration, DateTime, UTC};
 
 /// An item scheduled for delayed execution.
-struct Schedule {
+struct Schedule<T> {
     /// The instant at which to execute.
     date: DateTime<UTC>,
 
-    /// The callback to execute.
-    cb: Box<FnMut() + Send>,
+    /// The schedule data.
+    data : T,
```

Comment:
In that case, could you mention that it can be used to store a callback, a message, or something else.